src/zhinst/toolkit/driver/nodes/readout.py

```diff
@@ -150,6 +150,7 @@ def write_integration_weights(
         weights: t.Union[Waveforms, dict],
         *,
         integration_delay: float = 0.0,
+        integration_length: int = None,
```

Comment:
Looks great, one minor thing:

Type hint `t.Optional[int]` signals that the value can also be `None`